examples/max_trials_callback.py

```diff
@@ -12,21 +12,10 @@
 from time import sleep
 
 import optuna
+from optuna._callbacks import MaxTrialsCallback

```

Comment:
We could import `MaxTrialsCallback` at the top of `optuna/__init__.py` or `optuna/study.py` to make `MaxTrialsCallback` public.

Reply:
Let's add the callback to the documentation (.rst) too. For instance https://github.com/optuna/optuna/blob/master/docs/source/reference/optuna.rst, if it's defined in `optuna/__init__.py`.

Reply:
`import optuna` didn't work, cuz then it couldn't find `optuna.study`. Stringed type hints worked. Didn't know you could do that!